### PR TITLE
expect: strengthen test

### PR DESCRIPTION
**Pull request by fxcoudert**
Trying to see what platforms are affected by https://github.com/Homebrew/homebrew-core/issues/122541

### CODE REVIEW

File: Formula/expect.rb [hunk 1]
```diff
@@ -74,5 +74,6 @@ def install
   test do
     assert_match "works", shell_output("echo works | #{bin}/timed-read 1")
     assert_equal "", shell_output("{ sleep 3; echo fails; } | #{bin}/timed-read 1 2>&1")
+    assert_match "Done", shell_output("echo 'exec true; puts Done' | #{bin}/expect")
```
Comment (SMillerDev) [inline Formula/expect.rb:77]:
```suggestion
    assert_match "Done", pipe_output("#{bin}/expect", "exec true; puts Done")
```

Comment (fxcoudert) [reply Formula/expect.rb]: I just copied the syntax above :)

**Comment by fxcoudert** [comment]
So it's specific to macOS 13 and only on ARM. That's weird.

**Comment by cho-m** [comment]
Can be seen with optimization level `-Os` (upstream default on macOS) or `-O2/-O3`. Goes away with `-O1/-O0`.

Also, I don't see an issue if we replace `-Wno-implicit-function-declaration` workaround with patch: https://core.tcl-lang.org/expect/tktview/0d5b33c00e5b4bbedb835498b0360d7115e832a0

**Comment by cho-m** [comment]
From patch, forcing an include (e.g. `ENV.append_to_cflags "-include sys/ioctl.h"`) causes segfault to go away, but not sure if this is actually related to issue.

Maybe a minor modification is causing us to get lucky on some undefined behavior.

**Comment by fxcoudert** [comment]
> Maybe a minor modification is causing us to get lucky on some undefined behavior

Or including the proper header gives the right prototype, and fixes a pointer alignment or something like that. Original patch did not work on linux, because it lacks `util.h`. Trying a second version.

**Comment by fxcoudert** [comment]
Network failure in audit.

**Pull request closed without merging**